splunk_add_on_ucc_framework/ucc_ui_lib/src/main/webapp/components/CheckBoxComponent.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Switch from '@splunk/react-ui/Switch';
+
+class CheckBoxComponent extends Component {
+    constructor(props) {
+        super(props);
+    }
+
+    handleClick = (e) => {
+        this.props.handleClick(this.props.id, e.target.value);
+    };
+
+    render() {
+        return (
+            <Switch
+                key={this.props.field}
+                value={this.props.value}
+                onClick={this.handleClick}
+                selected={this.props.value === 1 ? true : false}
+                appearance="checkbox"
+            >
+                {this.props.label}
+            </Switch>
+        );
+    }
+}
+
+CheckBoxComponent.propTypes = {
+    id: PropTypes.number.isRequired,
+    value: PropTypes.string,
+    handleClick: PropTypes.func.isRequired,
+    field: PropTypes.string,
+    label: PropTypes.string,
```

Comment:
I think, you won't need to handle **label**  as discussed.

Reply:
Updated